src/main/java/org/opensearch/flowframework/workflow/CreateIngestPipelineStep.java

```diff
@@ -187,12 +112,75 @@ public PlainActionFuture<WorkflowData> execute(
                     logger.error(errorMessage, e);
                     createIngestPipelineFuture.onFailure(new FlowFrameworkException(errorMessage, ExceptionsHelper.status(e)));
                 }
+            }
 
-            }, exception -> {
+            @Override
+            public void onFailure(Exception e) {
                 String errorMessage = "Failed to create ingest pipeline";
-                logger.error(errorMessage, exception);
-                createIngestPipelineFuture.onFailure(new FlowFrameworkException(errorMessage, ExceptionsHelper.status(exception)));
-            }));
+                logger.error(errorMessage, e);
+                createIngestPipelineFuture.onFailure(new FlowFrameworkException(errorMessage, ExceptionsHelper.status(e)));
+            }
+
+        };
+
+        Set<String> requiredKeys = Set.of(PIPELINE_ID, CONFIGURATIONS);
+
+        // currently, we are supporting an optional param of model ID into the various processors
+        Set<String> optionalKeys = Set.of(MODEL_ID);
+
+        try {
+            Map<String, Object> inputs = ParseUtils.getInputsFromPreviousSteps(
+                requiredKeys,
+                optionalKeys,
+                currentNodeInputs,
+                outputs,
+                previousNodeInputs,
+                params
+            );
+
+            String pipelineId = (String) inputs.get(PIPELINE_ID);
+            String configurations = (String) inputs.get(CONFIGURATIONS);
+
+            // Regex to find patterns like ${{deploy_openai_model.model_id}}
+            // We currently support one previous node input that fits the pattern of (step.input_to_look_for)
+            Pattern pattern = Pattern.compile("\\$\\{\\{([\\w_]+)\\.([\\w_]+)\\}\\}");
+            Matcher matcher = pattern.matcher(configurations);
+
+            StringBuffer result = new StringBuffer();
+            while (matcher.find()) {
+                // Params map contains params for previous node input (e.g: deploy_openai_model:model_id)
+                // Check first if the substitution is looking for the same key, value pair and if yes
+                // then replace it with the key value pair in the inputs map
+                if (params.containsKey(matcher.group(1)) && params.get(matcher.group(1)).equals(matcher.group(2))) {
+                    // Extract the key for the inputs (e.g., "model_id" from ${{deploy_openai_model.model_id}})
+                    String key = matcher.group(2);
+                    if (inputs.containsKey(key)) {
+                        // Replace the whole sequence with the value from the map
+                        matcher.appendReplacement(result, (String) inputs.get(key));
+                    }
+                }
+            }
+            matcher.appendTail(result);
```

Comment:
The existing parsing that gets you `inputs` probably already handles this format (and should actually work after the bug fix #559 ... this whole section here seems to duplicate that).  The only thing missing is that we may want more optional inputs based on searching the configuration.  You could probably take the `conditionallySubstitute()` method and modify it to find the keys and add them to the optional inputs prior to calling that.

Reply:
Moved the parsing into the ParseUtils. conditionallySubstitute(), lmk if this looks good and if its what you meant. I tested we can still do param substitution through rest calls or previous nodes input incase of pipeline configuration